### DISCUSSION TITLE
z.infer of z.object shouldn't be non-nullable by default?

### DISCUSSION THREAD

**Discussion by renatorib** [colinhacks/zod#723] [Q&A]
I was trying to figure out how to create non-nullable schemas reading the documentation, but what I found is that is it's supposed to be non-nullable by default (until you explicitly add .optional, what makes sense).
Sooo, what can I be doing wrong?

![image](https://user-images.githubusercontent.com/3277185/138343721-bfd15356-ab7a-4f10-a816-43edbe465dd1.png)

**Comment by scotttrinh** [comment]
This is typically a symptom of not having `strict` (or at least `strictNullChecks` I think?) in your `tsconfig.json`. This is mentioned in the [README](https://github.com/colinhacks/zod#installation) as part of the installation instructions, but is often overlooked!
